doc/modules/pipeline.rst

```diff
@@ -266,4 +269,60 @@ and ignored by setting to ``None``::
 .. topic:: Examples:
 
  * :ref:`sphx_glr_auto_examples_feature_stacker.py`
- * :ref:`sphx_glr_auto_examples_hetero_feature_union.py`
+
+
+.. _column_transformer:
+
+ColumnTransformer for heterogeneous data
+========================================
+
+Many datasets contain features of different types, say text, floats and dates,
+where each type of feature requires separate preprocessing.
+Often it is easiest to preprocess data before applying scikit-learn methods,
+for example using :ref:`pandas <http://pandas.pydata.org/>`__.
+The :class:`~sklearn.experimental.ColumnTransformer` is a convenient way to perform heterogeneous
+preprocessing on data columns within a scikit-learn pipeline (for example,
+when you want to adjust preprocessing parameters within a grid search).
+The :class:`~sklearn.experimental.ColumnTransformer` works on arrays and pandas dataframes.
+
+To each column, a different transformation can be applied, such as
+preprocessing or a specific feature extraction method::
+
+  >>> import pandas as pd
+  >>> X = pd.DataFrame(
+  ...     {'city': ['London', 'London', 'Paris', 'New York'],
+  ...      'title': ["His Last Bow", "How Watson Learned the Trick", "A Moveable Feast", "The Great Gatsby"]})
+
+For this data, we might want to apply a :class:`~sklearn.preprocessing.OneHotEncoder` to the
```

Comment:
When you do this, the displayed text will be "OneHotEncoder" instead of "sklearn.preprocessing.OneHotEncoder", making the text a bit shorter but keeping the link working.

In many places we certainly want the full name so people are aware of the module it is in, but here I found it suited to use the shorter one. But happy to change.